codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/XmlShapeDeserVisitor.java

```diff
@@ -0,0 +1,403 @@
+            writer.write("if err != nil { return err }");
+            writer.write("if done { break }");
+
+            MemberShape member = shape.getMember();
+            Shape target = context.getModel().expectShape(member.getTarget());
+            String serializedMemberName = getSerializedMemberName(member);
+            writer.openBlock("switch t.Name.Local {", "}", () -> {
+                writer.openBlock("case $S:", "", serializedMemberName, () -> {
+                    writer.write("var col $P", context.getSymbolProvider().toSymbol(target));
+
+                    // in case of nested collections we will have nested `member` element tags
+                    if (target.isListShape() || target.isSetShape()) {
+                        writer.write("memberDecoder := smithydecoding.NewXMLNodeDecoder(decoder.Decoder, t)");
+                        writer.openBlock("for {", "}", () -> {
+                            writer.write("t, done, err = memberDecoder.Token()");
+                            writer.write("if err != nil { return err }");
+                            writer.write("if done { break }");
+
+                            CollectionShape targetAsCollection = getCollectionShape(target);
+                            MemberShape targetMember = targetAsCollection.getMember();
+                            Symbol targetMemberSymbol = symbolProvider.toSymbol(targetMember);
+                            Shape nestedTarget = context.getModel().expectShape(targetMember.getTarget());
+                            String serializedTargetMemberName = getSerializedMemberName(targetMember);
+
+                            writer.openBlock("switch t.Name.Local {", "}", () -> {
+                                writer.openBlock("case $S:", "", serializedTargetMemberName, () -> {
+                                    writer.write("originalDecoder := decoder");
+                                    writer.write("decoder = smithydecoding.NewXMLNodeDecoder(memberDecoder.Decoder, t)");
+                                    writer.openBlock("for {", "}", () -> {
+                                        writer.write("var mv $P", targetMemberSymbol);
+                                        nestedTarget.accept(getMemberDeserVisitor(targetMember, "mv", FunctionalUtils.alwaysTrue()));
+                                        writer.write("col = append(col, mv)");
+                                    });
+                                });
+                                writer.write("decoder = originalDecoder");
+                            });
+                        });
+                    } else {
+                        target.accept(getMemberDeserVisitor(member, "col", FunctionalUtils.alwaysTrue()));
+                    }
+                    writer.write("sv = append(sv, col)");
+                });
+
+                writer.openBlock("default:", "", () -> {
+                    writer.write("// Do nothing and ignore");
```

Comment:
You should use `writeDocs`. Also if you're going to write a comment here, I would say more about what's happened. This is a case of an unexpected tag right?

Reply:
Yes, the default can also be an additional element received in response which is not modeled.